components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/notification/mgt/NotificationManagementServiceImplTests.java

```diff
@@ -98,13 +104,66 @@ private RegistryService getRegistryService() throws RegistryException {
 
     @Test(description = "Add notifications using addNotification method and check whether it returns true.")
     public void addNotification() throws Exception {
```

Comment:
This description only talks about the successful notification. In the second part of the method, we are testing negative scenarios as well.